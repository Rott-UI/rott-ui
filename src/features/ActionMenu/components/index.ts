export * from './ActionMenuComponent'
export * from './ActionMenuHeaderComponent'
