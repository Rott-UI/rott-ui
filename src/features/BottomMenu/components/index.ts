export * from './BottomMenu'
export * from './BottomMenuItem'
