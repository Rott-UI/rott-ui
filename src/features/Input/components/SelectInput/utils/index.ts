export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
