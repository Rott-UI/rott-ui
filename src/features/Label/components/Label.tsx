// React Imports
import {FC, PropsWithChildren} from 'react'

// React Native Imports
import {Text, TextProps, StyleSheet} from 'react-native'

// Component Imports
import {LabelStyles} from '../styles/Label.style'
import {CommonUiProps} from '../../models'
import {FontFamily, FontWeight} from '../models'

interface LabelProps extends TextProps, CommonUiProps, PropsWithChildren {
  textCenter?: boolean
  fontWeight?: FontWeight
  fontFamily?: FontFamily
  letterSpacing?: number
}

export const Label: FC<LabelProps> = ({
  fontSize = 'md',
  variant = 'black',
  textCenter,
  fontWeight,
  fontFamily,
  letterSpacing = undefined,
  color,
  style,
  children,
  ...props
}) => {
  return (
    <Text
      style={StyleSheet.flatten([
        LabelStyles({
          textCenter,
          flex: 1,
          fontSize,
          variant,
          fontWeight,
          fontFamily,
          letterSpacing,
          color,
          includeLatterSpacing: true,
          ...props,
        }).defaultLabelStyle,
        style,
      ])}
      {...props}>
      {children}
    </Text>
  )
}
