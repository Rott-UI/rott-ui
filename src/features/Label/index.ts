export * from './components'
export * from './models'
export * from './utils'
export * from './styles'
