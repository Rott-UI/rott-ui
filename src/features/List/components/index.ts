export * from './List'
export * from './FavoriteSwipeComponent'
