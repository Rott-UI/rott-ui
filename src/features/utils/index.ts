export * from './commonUiTestExtension'
export * from './commonUiStyleProperties'
