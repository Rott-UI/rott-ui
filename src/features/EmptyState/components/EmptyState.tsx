// React Imports
import React, {FC} from 'react'

// Component Imports
import {EmptyStateProps} from '../models'
import {Item} from '../../Item'
import {Content} from '../../Content'
import {Image} from '../../Image'
import {Label} from '../../Label'

// Util and Lib Imports
import {display} from '../../../utils'

/**
 * Result - Islem Sonucu Ekrani
 *
 * @param name Image alanı
 * @param description Açıklama başlık alanı
 
 * @returns Sonuc Ekrani Renderlanır
 *
 */

export const EmptyState: FC<EmptyStateProps> = ({name, description}) => {
  return (
    <Content flex={1} size='full'>
      <Item justifyContentCenter alignItemsCenter marginTop={50} height={display.heightPixel(230)}>
        <Image
          name={name}
          resizeMode='contain'
          width={152}
          height={152}
          marginBottom={32}
          testID='empty-state-image-test-id'
        />

        {description && (
          <Item>
            {typeof description === 'string' && (
              <Label
                marginBottom={8}
                fontSize='lg'
                fontFamily='Markpro-Medium'
                textCenter
                variant='white'
                testID='empty-state-desc-test-id'>
                {description}
              </Label>
            )}

            {typeof description !== 'string' && description}
          </Item>
        )}
      </Item>
    </Content>
  )
}
