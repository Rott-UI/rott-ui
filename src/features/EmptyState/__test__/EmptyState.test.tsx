// React Imports
import React from 'react'

// Component Imports
import {EmptyState} from '../components'
import {Label} from '../../Label'

// Util and Lib Imports
import {formatMessage, render} from '../../../utils'

describe('EmptyState -> Custom Component', () => {
  const testId = {
    emptyStateImageTestId: 'empty-state-image-test-id',
    emptyStateDescTestId: 'empty-state-desc-test-id',
  }
  const testTexts = {
    description: formatMessage('TEST'),
  }

  it('ilk render anında snapshot ile eşleşmeli', () => {
    const {description} = testTexts
    const renderedEmptyState = render(<EmptyState name='EMPTY_PHONE' description={description} />)

    expect(renderedEmptyState).toMatchSnapshot()
  })

  it('name propertysi EMPTY ile başlamalı', () => {
    const {description} = testTexts
    const {emptyStateImageTestId} = testId
    const {getByTestId} = render(<EmptyState name='EMPTY_PHONE' description={description} />)

    const imageElement = getByTestId(emptyStateImageTestId)

    expect(imageElement.props.source.testUri).toMatch(/empty-state/gim)
  })

  it('description verilen değerler ile ekranda gözükmeli', () => {
    const {description} = testTexts
    const {getByText} = render(<EmptyState name='EMPTY_PHONE' description={description} />)

    const descriptionElement = getByText(description)

    expect(descriptionElement).toBeOnTheScreen()
  })

  it('description verilmediği zaman ekranda gözükmemeli', () => {
    const {emptyStateDescTestId} = testId
    const {queryByTestId} = render(<EmptyState name='EMPTY_PHONE' />)

    const descriptionElement = queryByTestId(emptyStateDescTestId)

    expect(descriptionElement).not.toBeOnTheScreen()
  })

  it('description yerine ReactNode olarak verildiğindede ekranda gösterilmeli', () => {
    const {description} = testTexts
    const {getByTestId} = render(
      <EmptyState
        name='EMPTY_PHONE'
        description={<Label testID='custom-label-test-id'>{description}</Label>}
      />
    )

    const descriptionElement = getByTestId('custom-label-test-id')

    expect(descriptionElement).toBeOnTheScreen()
  })
})
