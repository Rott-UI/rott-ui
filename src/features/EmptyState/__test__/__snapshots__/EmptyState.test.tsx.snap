// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState -> Custom Component ilk render anında snapshot ile eşleşmeli 1`] = `
<View
  alignItemsCenter={true}
  justifyContentCenter={true}
  style={
    {
      "alignItems": "center",
      "backgroundColor": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "column",
      "flexGrow": undefined,
      "gap": undefined,
      "height": undefined,
      "justifyContent": "center",
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
>
  <Image
    height={152}
    resizeMode="contain"
    source={
      {
        "testUri": "../../../src/assets/images/empty-state/phone.png",
      }
    }
    style={
      {
        "backgroundColor": undefined,
        "borderRadius": 50,
        "bottom": undefined,
        "flexGrow": undefined,
        "gap": undefined,
        "height": 293,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": 293,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "resizeMode": "contain",
        "right": undefined,
        "top": undefined,
        "width": 293,
        "zIndex": undefined,
      }
    }
    testID="empty-state-image-test-id"
    width={152}
  />
  <View
    alignItemsCenter={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": undefined,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
  >
    <Text
      marginBottom={8}
      style={
        {
          "backgroundColor": undefined,
          "bottom": undefined,
          "color": "rgba(255, 255, 255, 1)",
          "flexGrow": undefined,
          "fontFamily": "Markpro-Medium",
          "fontSize": 18,
          "fontWeight": undefined,
          "gap": undefined,
          "height": undefined,
          "left": undefined,
          "letterSpacing": undefined,
          "marginBottom": 16,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "textAlign": "center",
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
      testID="empty-state-desc-test-id"
    >
      Test
    </Text>
  </View>
</View>
`;
